### PR TITLE
Clarify setup steps

### LINKED ISSUE #10: Describe how to do initial setup in docs
Reported by Tommy Fisher on Slack.

It turns out `Nerves.Network.setup` isn’t meant to be supervised like `Nerves.Networking` was.
Since it returns an `:ok` tuple, its supervisor won't restart it, however, the `SystemRegistry` removed its info when the process died (since it was a transient `Task`).

Update the docs with a usage example so people know not to run it as a transient supervised `Task`.

### PR DESCRIPTION
**Pull request by tokafish**
Resolves #10

### CODE REVIEW

**Comment by fhunleth** [comment]
Thanks for sending this! In retrospect, I can't imagine how many people probably got tricked by the change to SystemRegistry and the requirement to "own" the configuration that you set. This will certainly help a lot of people.

File: README.md [hunk 1]
```diff
@@ -4,6 +4,26 @@
...
+In your firmware's `mix.exs`, add `nerves_network` to your deps for your system target:
+
+```elixir
+def deps(target) do
+  [ system(target),
+    {:nerves_network, "~> 0.3"}
+  ]
+```
+
+Also add it to your application:
```
Comment (fhunleth) [inline README.md]: This part isn't needed. Elixir 1.4 figures this out automatically.

Comment (tokafish) [inline README.md]: Thanks! I updated the PR.

**Comment by bdubaut** [comment]
@tokafish @fhunleth maybe [this section](https://github.com/nerves-project/nerves_network#setup) of the README could be updated in a similar manner?

**Comment by fhunleth** [comment]
@bdubaut Definitely a good idea. See #13.

**Pull request merged**